addons/payment_buckaroo/models/payment_transaction.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from werkzeug import urls
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+from odoo.addons.payment_buckaroo.controllers.main import BuckarooController
+
+_logger = logging.getLogger(__name__)
+
+
+def _normalize_dataset(data):
+    """ Set all keys of a dictionary to uppercase.
+
+    As Buckaroo parameters names are case insensitive, we can convert everything to upper case to
+    easily detected the presence of a parameter by checking the uppercase key only.
+
+    :param dict data: The dictionary whose keys must be set to uppercase
+    :return: A copy of the original data with all keys set to uppercase
+    :rtype: dict
+    """
+    return {key.upper(): val for key, val in data.items()}
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    # Buckaroo status codes
+    _pending_tx_status = [790, 791, 792, 793]
+    _valid_tx_status = [190]
+    _cancel_tx_status = [890, 891]
+
+    def _get_specific_rendering_values(self, processing_values):
+        if self.provider != 'buckaroo':
+            return super()._get_specific_rendering_values(processing_values)
+
+        buckaroo_tx_values = dict(processing_values)
+        return_url = urls.url_join(self.acquirer_id._get_base_url(), BuckarooController._return_url)
+        buckaroo_tx_values.update({
+            'tx_url': self.acquirer_id._buckaroo_get_api_url(),
+            'Brq_websitekey': self.acquirer_id.buckaroo_website_key,
+            'Brq_amount': processing_values['amount'],
+            'Brq_currency': self.currency_id.name,
+            'Brq_invoicenumber': processing_values['reference'],
+            'Brq_return': return_url,
+            'Brq_returncancel': return_url,
+            'Brq_returnerror': return_url,
+            'Brq_returnreject': return_url,
```

Comment:
Alphabetical order of inputs (or following the API documentation order ?)

Reply:
doc order